Plugins/Flow.Launcher.Plugin.Program/Programs/UWP.cs

```diff
@@ -601,75 +601,94 @@ internal string LogoPathFromUri(string uri)
                     {
-                        // scale factors on win10: https://docs.microsoft.com/en-us/windows/uwp/controls-and-patterns/tiles-and-notifications-app-assets#asset-size-tables,
-                        {
-                            PackageVersion.Windows10, new List<int>
-                            {
-                                100,
-                                125,
-                                150,
-                                200,
-                                400
-                            }
-                        },
-                        {
-                            PackageVersion.Windows81, new List<int>
-                            {
-                                100,
-                                120,
-                                140,
-                                160,
-                                180
-                            }
-                        },
-                        {
-                            PackageVersion.Windows8, new List<int>
-                            {
-                                100
-                            }
-                        }
-                    };
+                        return path; // shortcut, avoid enumerating files
+                    }
 
-                    if (scaleFactors.ContainsKey(Package.Version))
+                    var logoNamePrefix = Path.GetFileNameWithoutExtension(uri); // e.g Square44x44
+                    var logoDir = Path.GetDirectoryName(path);  // e.g ..\..\Assets
+                    if (String.IsNullOrEmpty(logoNamePrefix) || String.IsNullOrEmpty(logoDir) || !Directory.Exists(logoDir))
                     {
-                        foreach (var factor in scaleFactors[Package.Version])
-                        {
-                            paths.Add($"{prefix}.scale-{factor}{extension}");
-                        }
+                        // Known issue: Edge always triggers it since logo is not at uri
+                        ProgramLogger.LogException($"|UWP|LogoPathFromUri|{Package.Location}" +
+                           $"|{UserModelId} can't find logo uri for {uri} in package location (logo name or directory not found): {Package.Location}", new FileNotFoundException());
+                        return string.Empty;
                     }
 
-                    var selected = paths.FirstOrDefault(File.Exists);
+                    var files = Directory.EnumerateFiles(logoDir);
+
+                    // Just ignore all qulifiers
+                    // select like logo[xxx].png
+                    // https://learn.microsoft.com/en-us/windows/uwp/app-resources/tailor-resources-lang-scale-contrast
+                    var selected = files.FirstOrDefault(file => 
+                        Path.GetFileName(file).StartsWith(logoNamePrefix) && extension == Path.GetExtension(file)
+                    );
+
+                    //var scaleFactors = new Dictionary<PackageVersion, List<int>>
```

Comment:
why do we remove them?

Reply:
I can't find any guarantee in the doc that scale factors comes last to the qualifiers (see [here](https://learn.microsoft.com/en-us/windows/uwp/app-resources/images-tailored-for-scale-theme-contrast)). And we don't acutally need scale factor in our plugin.

Reply:
true, let's remove them. We shall fetch all asset information in the manifest file.